src/pages/storage/MigrateVolumeBtn.tsx

```diff
@@ -0,0 +1,137 @@
+import { FC, useState } from "react";
+import { ActionButton, Icon } from "@canonical/react-components";
+import usePortal from "react-useportal";
+import { useQueryClient } from "@tanstack/react-query";
+import { queryKeys } from "util/queryKeys";
+import { useEventQueue } from "context/eventQueue";
+import ItemName from "components/ItemName";
+import { useToastNotification } from "context/toastNotificationProvider";
+import { LxdStorageVolume } from "types/storage";
+import MigrateVolumeModal from "./MigrateVolumeModal";
+import { migrateStorageVolume } from "api/storage-pools";
+import { useLocation, useNavigate } from "react-router-dom";
+
+interface Props {
+  storageVolume: LxdStorageVolume;
+  project: string;
+  classname?: string;
+  onClose?: () => void;
+}
+
+const MigrateVolumeBtn: FC<Props> = ({
+  storageVolume,
+  project,
+  classname,
+  onClose,
+}) => {
+  const eventQueue = useEventQueue();
+  const toastNotify = useToastNotification();
+  const { openPortal, closePortal, isOpen, Portal } = usePortal();
+  const queryClient = useQueryClient();
+  const navigate = useNavigate();
+  const [isVolumeLoading, setVolumeLoading] = useState(false);
+  const location = useLocation();
+
+  const handleSuccess = (
+    newTarget: string,
+    storageVolume: LxdStorageVolume,
+  ) => {
+    toastNotify.success(
+      <>
+        Volume <ItemName item={{ name: storageVolume.name }} bold />{" "}
+        successfully migrated to pool{" "}
+        <ItemName item={{ name: newTarget }} bold />
+      </>,
+    );
+
+    const oldVolumeUrl = `/ui/project/${storageVolume.project}/storage/pool/${storageVolume.pool}/volumes/${storageVolume.type}/${storageVolume.name}`;
+    const newVolumeUrl = `/ui/project/${storageVolume.project}/storage/pool/${newTarget}/volumes/${storageVolume.type}/${storageVolume.name}`;
+    if (location.pathname.startsWith(oldVolumeUrl)) {
```

Comment:
```suggestion
    if (window.location.pathname.startsWith(oldVolumeUrl)) {
```
The location hook does not give us the current url at this point in time. So I do get redirected back to the new volume page, also in case I navigate quickly to the instances page after starting the migration process. But we can use the `window.location.pathname` instead and this should be fine.